source/com.microsoft.tfs.client.eclipse.ui.egit/src/com/microsoft/tfs/client/eclipse/ui/egit/protocolhandler/VstsInfoClientHandler.java

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the repository root.
+
+package com.microsoft.tfs.client.eclipse.ui.egit.protocolhandler;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.UUID;
+
+import com.microsoft.alm.client.HttpMethod;
+import com.microsoft.alm.client.TeeClientHandler;
+import com.microsoft.alm.client.VssRestClientHandler;
+import com.microsoft.alm.client.VssRestRequest;
+import com.microsoft.alm.visualstudio.services.webapi.ApiResourceLocationCollection;
+import com.microsoft.alm.visualstudio.services.webapi.ApiResourceVersion;
+import com.microsoft.tfs.client.common.ui.config.UIClientConnectionAdvisor;
+import com.microsoft.tfs.core.TFSTeamProjectCollection;
+import com.microsoft.tfs.core.httpclient.Credentials;
+import com.microsoft.tfs.core.httpclient.HttpClient;
+import com.microsoft.tfs.core.util.URIUtils;
+import com.microsoft.tfs.util.Check;
+
+public class VstsInfoClientHandler extends TeeClientHandler implements VssRestClientHandler {
+
+    private final static UUID UUID_EMPTY = UUID.fromString("00000000-0000-0000-0000-000000000000"); //$NON-NLS-1$
+
+    public VstsInfoClientHandler(final String targetURL, final Credentials credentials) {
+        super(getFakeCollection(targetURL, credentials).getHTTPClient());
+    }
+
+    public VstsInfoClientHandler(HttpClient httpClient) {
+        super(httpClient);
+    }
+
+    /**
+     * The checkConnection method is not applicable for this client.
+     */
+    @Override
+    @Deprecated
```

Comment:
The methods are defined in the base class and are not deprecated there (used in other derived classes). Just in this special derived class we shouldn't use them. That's actually our internal waring for TEE developers.